mmv1/third_party/terraform/services/apigee/resource_apigee_target_server_test.go

```diff
@@ -0,0 +1,362 @@
+}
+
+resource "google_compute_network" "apigee_network" {
+  name       = "apigee-network"
+  project    = google_project.project.project_id
+  depends_on = [google_project_service.compute]
+}
+
+resource "google_compute_global_address" "apigee_range" {
+  name          = "apigee-range"
+  purpose       = "VPC_PEERING"
+  address_type  = "INTERNAL"
+  prefix_length = 16
+  network       = google_compute_network.apigee_network.id
+  project       = google_project.project.project_id
+}
+
+resource "google_service_networking_connection" "apigee_vpc_connection" {
+  network                 = google_compute_network.apigee_network.id
+  service                 = "servicenetworking.googleapis.com"
+  reserved_peering_ranges = [google_compute_global_address.apigee_range.name]
+  depends_on              = [google_project_service.servicenetworking]
+}
+
+resource "google_apigee_organization" "apigee_org" {
+  analytics_region   = "us-central1"
+  project_id         = google_project.project.project_id
+  authorized_network = google_compute_network.apigee_network.id
+  depends_on         = [
+    google_service_networking_connection.apigee_vpc_connection,
+    google_project_service.apigee,
+  ]
+}
+
+resource "google_apigee_environment" "apigee_environment" {
+  org_id       = google_apigee_organization.apigee_org.id
+  name         = "tf-test%{random_suffix}"
+  description  = "Apigee Environment"
+  display_name = "environment-1"
+}
+
+resource "google_apigee_env_keystore" "apigee_environment_keystore" {
+	name   = "tf-test-keystore%{random_suffix}"
+  env_id = google_apigee_environment.apigee_environment.id
+}
+   
+resource "google_apigee_keystores_aliases_key_cert_file" "apigee_test_alias" {
+	alias       = "tf-test-alias%{random_suffix}"
+	org_id      = google_apigee_organization.apigee_org.name
+	environment = google_apigee_environment.apigee_environment.name
+	cert        = file("./test-fixtures/apigee_keystore_alias_test_cert.pem")
+	key         = sensitive(file("./test-fixtures/apigee_keystore_alias_test_key.pem"))
+	password    = sensitive("password")
+	keystore    = google_apigee_env_keystore.apigee_environment_keystore.name
+}  
+   
+resource "google_apigee_target_server" "apigee_target_server"{
+	env_id                    = google_apigee_environment.apigee_environment.id
+	is_enabled                = true
+	name                      = "tf-test-target-server%{random_suffix}"
+	host                      = "host.test.com"
+	port                      = 443
+	protocol                  = "HTTP"
+	s_sl_info{
+	 enabled                  = true
+	 ciphers                  = ["TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA38411"]
+	 client_auth_enabled      = true
+	 ignore_validation_errors = true
+	 key_alias                = google_apigee_keystores_aliases_key_cert_file.apigee_test_alias.alias
+	 key_store                = google_apigee_env_keystore.apigee_environment_keystore.name
+	 protocols                = ["TLSv1.1"]
```

Comment:
Something that can cause problems with lists in resources is if the API changes the order of elements versus what a user has in their configuration. If possible, could you update one of the steps in this test so that the lists include more than one element? Having values out of lexicographical order in the config, e.g. `["TLSv1.2", "TLSv1.1"]`, can help identify potential re-ordering done by the API and the potential for permanent diffs being identified by Terraform

Reply:
test added. Thanks.